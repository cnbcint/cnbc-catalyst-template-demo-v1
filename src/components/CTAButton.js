import React, { useContext } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { breakpoints } from "../helpers/Layout";
import { LayoutContext } from "./Main";

const Link = styled.a`
  color: ${(props) => props.colors.copy};
  padding: 0.5rem 3rem;
  display: block;
  text-align: center;
  background-color: ${(props) => props.colors.primary};
  @media (min-width: ${breakpoints.mobile}) {
    display: inline-block;
  }

  text-decoration: none;
  &:hover {
    background-color: ${(props) => props.colors.secondary};
  }
`;
/**
 * Call To Action Button this element will use primary color from Theme.json and hover state secondary color from theme.json
 */
export default function CTAButton(props) {
  const layout = useContext(LayoutContext);

  return (
    <Link
      href={props.link}
      target={props.target}
      className={props.className}
      colors={layout.colors}
    >
      {props.children}
    </Link>
  );
}
CTAButton.defaultProps = {
  className: "",
};
CTAButton.propTypes = {
  className: PropTypes.string,
  link: PropTypes.string.isRequired,
  target: PropTypes.oneOf(["_self", "_blank"]),
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.element,
  ]).isRequired,
};
