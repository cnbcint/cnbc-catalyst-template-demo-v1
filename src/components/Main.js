import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { getMargin, breakpoints } from "../helpers/Layout";
import { ModalContext } from "../components/Modal";

export const LayoutContext = React.createContext({
  maxWidth: "500px",
  colors: null,
});

const Container = styled.div`
  display: block;
  width: 100%;
  background-color: ${(props) => props.colors.background};
  > * {
    font-size: 1.125rem;
    line-height: 140%;
    color: ${(props) => props.colors.copy};
  }
  hr {
    // border-color: ${(props) => props.primaryColor};
    border-color: ${(props) => props.colors.copy};
    border-style: solid;
    display: inline-block;
    width: 100%;
    // margin-top: ${getMargin(1)};
    margin-bottom: ${getMargin(1)};
    @media (min-width: ${breakpoints.mobile}) {
      // margin-top: ${getMargin(2)};
      margin-bottom: ${getMargin(2)};
    }
  }
  img {
    width: 100%;
    display: block;
    margin-top: ${getMargin(2)};
    margin-bottom: ${getMargin(2)};
  }
  p {
    margin: 1rem auto !important;
  }
  ul {
    list-style: disc !important;
    margin-top: ${getMargin(1)} !important;
    margin-bottom: ${getMargin(1)} !important;
    li {
      list-style: disc !important;
      margin: ${getMargin(1)} !important;
      margin-left: ${getMargin(2)} !important;
    }
  }
`;
export default function Main(props) {
  return (
    <LayoutContext.Provider
      value={{ maxWidth: props.maxWidth, colors: props.colors }}
    >
      <ModalContext.Provider value={{ modals: {} }}>
        <Container
          colors={props.colors}
          primaryColor={props.colors.primary}
          className={props.className}
          style={props.style}
        >
          {props.children}
        </Container>
      </ModalContext.Provider>
    </LayoutContext.Provider>
  );
}
Main.defaultProps = {
  maxWidth: "900px",
  colors: { primary: "black", secondary: "gray", tertiary: "red" },
};
Main.propTypes = {
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.element,
  ]).isRequired,
  maxWidth: PropTypes.string,
  colors: PropTypes.shape({
    primary: PropTypes.string,
    secondary: PropTypes.string,
    tertiary: PropTypes.string,
    background: PropTypes.string,
    copy: PropTypes.string,
  }),
  style: PropTypes.object,
};
