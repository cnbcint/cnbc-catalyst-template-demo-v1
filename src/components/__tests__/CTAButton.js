import React from "react";
import CTAButton from "../CTAButton";
import Main from "../Main";
import { render } from "@testing-library/react";
import "jest-styled-components";

const Theme = {
  maxWidth: "900px",
  colors: {
    primary: "orange",
    secondary: "green",
    tertiary: "orange",
    background: "white",
    copy: "black",
    links: "blue",
  },
};

describe("CTAButton", () => {
  it("Renders correctly", () => {
    const { container } = render(
      <Main {...Theme}>
        <CTAButton link="http://test.comu">Testing Content</CTAButton>
      </Main>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("Renders the correct content", () => {
    const { container } = render(
      <Main {...Theme}>
        <CTAButton link="http://test.comu">Testing Content</CTAButton>
      </Main>
    );
    expect(container.firstChild.textContent).toMatch("Testing Content");
  });

  it("Passes the class name", () => {
    const { container } = render(
      <Main {...Theme}>
        <CTAButton className="testClass" link="http://test.comu">
          Testing Content
        </CTAButton>
      </Main>
    );
    expect(
      container.firstChild.firstChild.classList.contains("testClass")
    ).toBe(true);
  });
});
