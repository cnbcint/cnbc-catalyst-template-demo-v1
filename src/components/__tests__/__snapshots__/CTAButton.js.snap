// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTAButton Renders correctly 1`] = `
.c0 {
  display: block;
  width: 100%;
  background-color: white;
}

.c0 > * {
  font-size: 1.125rem;
  line-height: 140%;
  color: black;
}

.c0 hr {
  border-color: orange;
  border-style: solid;
  display: inline-block;
  width: 100%;
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.c0 img {
  width: 100%;
  display: block;
  margin-top: 2rem;
  margin-bottom: 2rem;
}

.c0 p {
  margin-top: 1rem !important;
  margin-bottom: 1rem !important;
}

.c0 ul {
  list-style: disc !important;
  margin-top: 1rem !important;
  margin-bottom: 1rem !important;
}

.c0 ul li {
  list-style: disc !important;
  margin: 1rem !important;
  margin-left: 2rem !important;
}

.c1 {
  color: black;
  padding: 0.5rem 3rem;
  display: block;
  text-align: center;
  background-color: orange;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:hover {
  background-color: green;
}

@media (min-width:768px) {
  .c0 hr {
    margin-top: 2rem;
    margin-bottom: 2rem;
  }
}

@media (min-width:768px) {
  .c1 {
    display: inline-block;
  }
}

<div
  class="c0"
>
  <a
    class="c1"
    href="http://test.comu"
  >
    Testing Content
  </a>
</div>
`;
